src/components/Elements.test.tsx

```diff
@@ -0,0 +1,385 @@
+      <Elements stripe={mockStripe}>{children}</Elements>
+    );
+
+    const {result} = renderHook(() => useElements(), {wrapper});
+
+    expect(result.current).toBe(mockElements);
+  });
+
+  test('only creates elements once', () => {
+    const TestComponent = () => {
+      const _ = useElements();
+      return <div />;
+    };
+
+    render(
+      <Elements stripe={mockStripe}>
+        <TestComponent />
+      </Elements>
+    );
+
+    expect(mockStripe.elements).toHaveBeenCalledTimes(1);
+  });
+
+  // TODO(christopher): support Strict Mode first
+  // eslint-disable-next-line jest/no-disabled-tests
+  test.skip('only creates elements once in Strict Mode', () => {
+    const TestComponent = () => {
+      const _ = useElements();
+      return <div />;
+    };
+
+    render(
+      <React.StrictMode>
+        <Elements stripe={mockStripe}>
+          <TestComponent />
+        </Elements>
+      </React.StrictMode>
+    );
+
+    expect(mockStripe.elements).toHaveBeenCalledTimes(1);
+  });
+
+  test('injects stripe with the useStripe hook', () => {
+    const wrapper = ({children}: any) => (
+      <Elements stripe={mockStripe}>{children}</Elements>
+    );
+
+    const {result} = renderHook(() => useStripe(), {wrapper});
+
+    expect(result.current).toBe(mockStripe);
+  });
+
+  test('provides elements and stripe with the ElementsConsumer component', () => {
+    expect.assertions(2);
+
+    render(
+      <Elements stripe={mockStripe}>
+        <ElementsConsumer>
+          {(ctx) => {
+            expect(ctx.elements).toBe(mockElements);
+            expect(ctx.stripe).toBe(mockStripe);
+
+            return null;
+          }}
+        </ElementsConsumer>
+      </Elements>
+    );
+  });
+
+  test('provides elements and stripe with the ElementsConsumer component in Strict Mode', () => {
+    expect.assertions(2);
+
+    render(
+      <Elements stripe={mockStripe}>
+        <ElementsConsumer>
+          {(ctx) => {
+            expect(ctx.elements).toBe(mockElements);
+            expect(ctx.stripe).toBe(mockStripe);
+
+            return null;
+          }}
+        </ElementsConsumer>
+      </Elements>
+    );
+  });
+
+  test('provides given stripe instance on mount', () => {
+    const TestComponent = () => {
+      const stripe = useStripe();
+
+      if (!stripe) {
+        throw new Error('Stripe instance is null');
+      }
+
+      return null;
+    };
+
+    expect(() => {
+      render(
+        <Elements stripe={mockStripe}>
+          <TestComponent />
+        </Elements>
+      );
+    }).not.toThrow('Stripe instance is null');
+  });
+
+  test('allows a transition from null to a valid Stripe object', () => {
+    let stripeProp: any = null;
+    const wrapper = ({children}: any) => (
+      <Elements stripe={stripeProp}>{children}</Elements>
+    );
+
+    const {result, rerender} = renderHook(() => useElements(), {wrapper});
+    expect(result.current).toBe(null);
+
+    stripeProp = mockStripe;
```

Comment:
Yeah, does feel a bit weird (though it is what's prescribed in the [React Hooks Testing Library](https://react-hooks-testing-library.com/usage/basic-hooks#props)). Because you expect that props should not be mutated outside of the React lifecycle, right?

Would this be preferable? 
```tsx
  test('allows a transition from null to a valid Stripe object', () => {
    const TestComponent = () => {
      const elements = useElements();

      if (!elements) {
        return null;
      }

      if (elements === mockElements) {
        return <>expected</>;
      }

      throw new Error(`unexpected useElements value: ${elements}`);
    };

    const {container, rerender} = render(
      <Elements stripe={null}>
        <TestComponent />
      </Elements>
    );

    expect(container).toBeEmpty();

    rerender(
      <Elements stripe={mockStripe}>
        <TestComponent />
      </Elements>
    );

    expect(container).toHaveTextContent('expected');
  });

```

Reply:
Updated this and the next, lemme know what you think.

Reply:
This actually seems too round-about. The way you had it before is fine after I get my head around it. Especially if the old way is the blessed way of testing.